src/Mod/Path/PathScripts/PathArray.py

```diff
@@ -155,54 +166,92 @@ def rotatePath(self, path, angle, centre):
 
         return newPath
 
-    def execute(self, obj):
-        if obj.Base:
-            if not obj.Base.isDerivedFrom("Path::Feature"):
-                return
-            if not obj.Base.Path:
-                return
-            if not obj.Base.ToolController:
-                return
+    def calculateJitter(self, obj, pos):
+        if random.randint(0,100) < obj.JitterPercent:
+            pos.x = pos.x + random.uniform(-obj.JitterMagnitude.x, obj.JitterMagnitude.y)
+            pos.y = pos.y + random.uniform(-obj.JitterMagnitude.y, obj.JitterMagnitude.y)
+            pos.z = pos.z + random.uniform(-obj.JitterMagnitude.z, obj.JitterMagnitude.z)
+        return pos
+
 
-            obj.ToolController = obj.Base.ToolController
+    def execute(self, obj):
+        if isinstance(obj.Base, list):
+            base = obj.Base
+        else:
+            base = [obj.Base]
+
+        if len(base)>0:
+            obj.ToolController = base[0].ToolController
+            for b in base:
+                if not b.isDerivedFrom("Path::Feature"):
+                    return
+                if not b.Path:
+                    return
+                if not b.ToolController:
+                    return
+                if b.ToolController != obj.ToolController:
+                    # this may be important if Job output is split by tool controller
+                    PathLog.warning('Arrays of paths having different tool controllers are handled according to the tool controller of the first path.')
```

Comment:
Not required for acceptance of this PR but if you're up for it, you could also check if the base feature is a dressup and allow for arrays of those as well.  Currently there's no way to make an array of dressed up operations.